crates/ruma-serde/src/canonical_json.rs

```diff
@@ -24,6 +24,7 @@ pub fn to_string<T: Serialize>(val: &T) -> Result<String, Error> {
 
 /// The set of possible errors when serializing to canonical JSON.
 #[derive(Debug)]
+#[non_exhaustive]
 pub enum Error {
```

Comment:
This particular enum seems like it will never be extended, and there might be some value in being able to exhaustively match it.